src/libsyncengine/reconciliation/syncoperation.h

```diff
@@ -67,6 +69,7 @@ class SyncOperation {
         ReplicaSide _targetSide = ReplicaSide::Unknown; // The side on which we will apply the operation
         bool _omit = false; // If true, apply change only in DB
         SyncName _newName; // New name on the replica on which we will apply the operation. Only for create and move operation
+        SyncPath _targetPath; // Relative path of the item to create. Only for local create operation.
```

Comment:
This would imply not returning a SyncPath& anymore but a SyncPath.
        [[nodiscard]] const SyncPath targetPath() const {
            if (_type == OperationType::Create && _targetSide == ReplicaSide::Local) {
                return _targetPath ? *_targetPath : SyncPath();
            } else {
                return nodePath(_targetSide);
            }
        }
I'm not sure it's a good idea to lose efficiency in order to gain genericity (bearing in mind that the else case is not used).

Reply:
As discussed offline, let's find a variable and function name that highlight the local and creative context in which the newly introduced members will be useful. In a second step, we should refactor `SyncOp` using inheritance to encapsulate operation specific data in a better way.